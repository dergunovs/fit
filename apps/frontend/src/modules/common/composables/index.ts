import { ref, computed, onMounted } from 'vue';
import { useRouter, useRoute } from 'vue-router';

import LayoutAdmin from '@/common/components/LayoutAdmin.vue';
import LayoutDefault from '@/common/components/LayoutDefault.vue';

export function useLayout() {
  const router = useRouter();
  const route = useRoute();

  const isLoaded = ref(false);

  const layoutComponent = computed(() => (route.meta.layout === 'admin' ? LayoutAdmin : LayoutDefault));

  onMounted(async () => {
    await router.isReady();
    isLoaded.value = true;
  });

  return {
    isLoaded,
    layoutComponent,
  };
}

export function useRouteId(title: string, isQuery?: boolean) {
  const route = useRoute();

  const id = computed(() => {
    const value = isQuery ? route.query[title]?.toString() : route.params[title].toString();

    return value || '';
  });

  return { id };
}
