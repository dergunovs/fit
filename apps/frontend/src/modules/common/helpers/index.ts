import { toast } from 'mhz-ui';

export async function copyToClipboard(text: string, toastText: string) {
  await navigator.clipboard.writeText(text);

  toast.success(toastText);
}
