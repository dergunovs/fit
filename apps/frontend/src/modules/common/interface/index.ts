import { TPostAuthLoginDTO } from 'fitness-tracker-contracts';
import { FunctionalComponent } from 'vue';

export interface INavItem {
  url: string;
  title: string;
  icon: FunctionalComponent;
}

export interface IOnSuccess {
  create?: (id?: string) => Promise<void>;
  update?: () => Promise<void>;
  delete?: () => Promise<void>;
  login?: (user: TPostAuthLoginDTO) => Promise<void>;
  setup?: () => Promise<void>;
  register?: () => Promise<void>;
}

export interface IPromptEvent extends Event {
  prompt: () => Promise<void>;
}

export interface IGetLatestVersionDTO {
  'dist-tags': { latest: string };
}
