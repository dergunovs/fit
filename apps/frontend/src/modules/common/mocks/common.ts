import { vi } from 'vitest';

export const spyCopyToClipboard = vi.spyOn(navigator.clipboard, 'writeText').mockResolvedValue();
