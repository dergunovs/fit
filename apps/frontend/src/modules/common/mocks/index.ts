export * from './helpers';
export * from './ui';
export * from './router';
export * from './layout';
export * from './pwa';
export * from './service';
