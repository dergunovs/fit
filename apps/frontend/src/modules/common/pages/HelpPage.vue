<template>
  <UiFlex column gap="16">
    <h1>{{ t('help') }}</h1>

    <div>
      <p>
        {{ t('sourceCode') }}:
        <a href="https://github.com/dergunovs/fit" target="_blank" rel="nofollow noreferrer noopener">Github</a>
      </p>

      <p>
        {{ t('videoLinks') }}:
        <a href="https://vkvideo.ru/playlist/15313879_11" target="_blank" rel="nofollow noreferrer noopener">VK</a>,
        <a href="https://youtube.com/@dergunovalexandr" target="_blank" rel="nofollow noreferrer noopener">Youtube</a>
      </p>

      <p>{{ t('emailLink') }}: mail@app-fit.ru</p>
    </div>

    <h3>{{ t('installation') }}</h3>

    <div>
      <b>RuStore(Android): </b>
      <a href="https://www.rustore.ru/catalog/app/com.appfit.dev" target="_blank" rel="nofollow noreferrer noopener">
        {{ t('installation') }}
      </a>
    </div>

    <div>
      <p v-for="(content, index) in tm('installText')" :key="index">
        {{ rt(content) }}
      </p>
    </div>

    <h3>{{ t('registration') }}</h3>

    <div>
      <p v-for="(content, index) in tm('registrationText')" :key="index">
        {{ rt(content) }}
      </p>
    </div>

    <h3>{{ t('activity.passing') }}</h3>

    <div>
      <p v-for="(content, index) in tm('activityPassingText')" :key="index">
        {{ rt(content) }}
      </p>
    </div>

    <UserFeedbackForm />
  </UiFlex>
</template>

<script setup lang="ts">
import { onMounted } from 'vue';
import { useI18n } from 'vue-i18n';
import { UiFlex } from 'mhz-ui';
import { scrollToTop } from 'mhz-helpers';

import UserFeedbackForm from '@/user/components/UserFeedbackForm.vue';

const { t, tm, rt } = useI18n();

onMounted(() => scrollToTop('main'));
</script>
