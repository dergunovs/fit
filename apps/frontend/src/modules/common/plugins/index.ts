import { createI18n, I18nOptions } from 'vue-i18n';

import ru from '@/common/locales/ru.json';
import en from '@/common/locales/en.json';

const messages = { en, ru } as const;

const options: I18nOptions = { legacy: false, locale: 'ru', messages };

// eslint-disable-next-line @typescript-eslint/no-type-alias
export type TMessageSchema = typeof en;

export const i18n = createI18n<false, typeof options>(options);
