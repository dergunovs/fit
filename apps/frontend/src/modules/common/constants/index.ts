import IconExercise from '@/common/icons/nav-exercise.svg?component';
import IconActivity from '@/common/icons/nav-activity.svg?component';
import IconEquipment from '@/common/icons/weight.svg?component';
import IconUser from '@/common/icons/nav-user.svg?component';

import { URL_USER } from '@/user/constants';
import { URL_EXERCISE } from '@/exercise/constants';
import { URL_ACTIVITY_ADMIN } from '@/activity/constants';
import { URL_EQUIPMENT } from '@/equipment/constants';
import { INavItem } from '@/common/interface';
import { URL_SETUP } from '@/auth/constants';

export const NAV_ITEMS: INavItem[] = [
  { _id: '1', url: URL_EXERCISE, title: 'Упражнения', icon: IconExercise },
  { _id: '2', url: URL_ACTIVITY_ADMIN, title: 'Занятия', icon: IconActivity },
  { _id: '3', url: URL_EQUIPMENT, title: 'Оборудование', icon: IconEquipment },
  { _id: '4', url: URL_USER, title: 'Пользователи', icon: IconUser },
];

export const UPDATE_BUTTON_TEXT = 'Обновить';
export const CREATE_BUTTON_TEXT = 'Создать';

export const URL_HOME = '/';
export const URL_ERROR = '/404';

export const GUEST_PAGES = [URL_HOME, URL_SETUP];
