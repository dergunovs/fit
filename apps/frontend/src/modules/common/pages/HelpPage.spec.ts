import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';

import HelpPage from './HelpPage.vue';

import { wrapperFactory } from '@/common/test';

let wrapper: VueWrapper<InstanceType<typeof HelpPage>>;

beforeEach(() => {
  wrapper = wrapperFactory(HelpPage);
});

enableAutoUnmount(afterEach);

describe('HelpPage', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(HelpPage)).toBeTruthy();
  });
});
