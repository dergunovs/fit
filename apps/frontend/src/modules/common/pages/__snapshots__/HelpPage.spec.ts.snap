// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelpPage > matches snapshot 1`] = `
"<div column="" gap="16">
  <h1>Помощь</h1>
  <div>
    <p>Исходный код: <a href="https://github.com/dergunovs/fit" target="_blank" rel="nofollow noreferrer noopener">Github</a></p>
    <p>Видео про разработку приложения: <a href="https://vkvideo.ru/playlist/15313879_11" target="_blank" rel="nofollow noreferrer noopener">VK</a>, <a href="https://youtube.com/@dergunovalexandr" target="_blank" rel="nofollow noreferrer noopener">Youtube</a></p>
    <p>Электронная почта для связи: mail@app-fit.ru</p>
  </div>
  <user-feedback-form-stub></user-feedback-form-stub>
  <h3>Установка</h3>
  <div><b>RuStore(Android): </b><a href="https://www.rustore.ru/catalog/app/com.appfit.dev" target="_blank" rel="nofollow noreferrer noopener">Установка</a></div>
  <div>
    <p>Приложение также можно установить через браузер (Google Chrome, в iOS - Safari):</p>
    <p>Android: зайдите на сайт app-fit.ru, войдите в меню браузера ⋮ и выберите 'Добавить на главный экран'.</p>
    <p>iOS: зайдите на сайт app-fit.ru, нажмите иконку 'Поделиться' и выберите 'Добавить на главный экран'.</p>
    <p>PC: зайдите на сайт app-fit.ru, в адресной строке браузера нажмите кнопку 'Установить'.</p>
  </div>
  <h3>Регистрация</h3>
  <div>
    <p>Зарегистрируйтесь, введя своё имя, почту и пароль. На почту придёт ссылка с подтверждением регистрации. Проверьте спам, если письма нет.</p>
    <p>Войдите в приложение. В профиле добавьте доступное для вас оборудование из списка. От этого зависит список упражнений, которые вы сможете выполнять. У некоторого оборудования есть выбор веса. Например, у вас могут быть гантели 2 и 3 кг.</p>
    <p>Если в упражнении возможен выбор веса или у вас несколько видов оборудования с весом, появится возможность выбора веса по-умолчания для каждого из таких упражнений. Сохраните данные, нажав кнопку 'Сохранить'.</p>
    <p>Вы можете добавить до 20 своих упражнений через форму в профиле пользователя. При создании упражнения добавьте его название, описание, задействованное оборудование и группы мышц.</p>
  </div>
  <h3>Прохождение занятия</h3>
  <div>
    <p>Добавьте упражнения с учётом количества повторов и доступного веса. Есть фильтр по группам мышц и названию упражнений.</p>
    <p>Можно добавлять сразу по 2 или 3 подхода. Если два последних упражнения отличаются, то появится ссылка на добавление сета из таких же упражнений.</p>
    <p>Если вы выполнили хотя бы одно занятие, то появится возможность повторить прошлое занятие в один клик. Также вы можете скопировать любое из выполненных ранее занятий через календарь на странице статистики, нажав на кнопку 'Скопировать' внизу всплывающего окна.</p>
    <p>Планируйте будущие занятия. Для этого нажмите на кнопку 'Добавить в календарь' при создании занятия. В появившемся календаре выберите дату и нажмите кнопку 'Сохранить занятие в календаре'. Затем в календаре вы можете начать занятие, причём не обязательно это делать в запланированный день.</p>
    <p>После формирования занятия нажмите кнопку 'Начать занятие'. Появится экран прохождения занятия. Нажмите 'Начать', чтобы запустить таймер текущего упражнения. Если вы чувствуете, что подход выполнен до отказа, то отметьте это. Вы можете скорректировать количество выполненных повторений. После этого нажмите кнопку 'Завершить'.</p>
    <p>После выполнения упражнений вы будете перенаправлены на страницу со статистикой. Каждое выполненное занятие в календаре окрашивается в цвет самых задействованных групп мышц. Чем больше группа мышц задействована, тем больше места она занимает. Также занятие в календаре можно удалить.</p>
    <p>Если вы не можете продолжать занятие, то можно завершить его досрочно, нажав по соответствующей кнопке. Использование таймера позволяет прогнозировать время выполнения будущих занятий, отслеживать среднее время отдыха, а также наглядно отображать занятия в статистике.</p>
  </div>
  <h3>Планируемая функциональность</h3>
  <div>
    <p>Приложение на раннем этапе разработки. Его функциональность будет существенно расширена. Тем не менее, вы уже можете им пользоваться, сохраняя данные своих тренировок. В планах внедрение следующей функциональности:</p>
    <ul>
      <li>корректировка веса, использованного в подходе;</li>
      <li>редактирование занятия в процессе его прохождения;</li>
      <li>тип пользователя 'Тренер', с возможностью планировать тренировки подопечных;</li>
      <li>оффлайн-режим;</li>
      <li>таймер времени отдыха;</li>
    </ul>
  </div>
  <h3>Известные проблемы</h3>
  <p>На iOS не работает запрет блокировки экрана при прохождении тренировки. Нужно разблокировать вручную или выставлять время блокировки экрана дольше вашего времени подхода.</p>
</div>"
`;
